### PR TITLE
Hdf5 1 12 Miscellaneous warnings fixes

### PR DESCRIPTION
**Pull request by jhendersonHDF**
Also cherry picks a few relevant commits from develop.

### CODE REVIEW

File: hl/src/H5PT.c [hunk 1]
```diff
@@ -29,7 +29,7 @@ static H5I_type_t H5PT_ptable_id_type = H5I_UNINIT;
 #define H5PT_HASH_TABLE_SIZE 64
 
 /* Packet Table private functions */
-static herr_t H5PT_free_id(void *id, void **_ctx);
+static herr_t H5PT_free_id(void *id);
```
Comment (jhendersonHDF) [inline hl/src/H5PT.c:32]: The ctx parameter was added to the public H5I_free_t typedef in 1.13 for async operations, but isn't in the 1.12 branch.

File: src/H5B2int.c [hunk 2]
```diff
@@ -52,9 +52,9 @@
 /********************/
 /* Local Prototypes */
 /********************/
-static herr_t H5B2__update_child_flush_depends(H5B2_hdr_t *hdr, unsigned depth,
-                                               const H5B2_node_ptr_t *node_ptrs, unsigned start_idx,
-                                               unsigned end_idx, void *old_parent, void *new_parent);
+static herr_t H5B2__update_child_flush_depends(H5B2_hdr_t *hdr, unsigned depth, H5B2_node_ptr_t *node_ptrs,
+                                               unsigned start_idx, unsigned end_idx, void *old_parent,
+                                               void *new_parent);
```
Comment (jhendersonHDF) [inline src/H5B2int.c:57]: Cherry-picked Dana's Btree const fixes

File: tools/lib/h5tools.c [hunk 3]
```diff
@@ -1843,15 +1843,21 @@ render_bin_output(FILE *stream, hid_t container, hid_t tid, void *_mem, hsize_t
                     hid_t        region_id    = H5I_INVALID_HID;
                     hid_t        region_space = H5I_INVALID_HID;
                     H5S_sel_type region_type;
+                    H5R_ref_t    tref;
```
Comment (jhendersonHDF) [inline tools/lib/h5tools.c:1846]: Cherry-picked Dave's reference misalignment fix

File: src/H5trace.c [hunk 4]
```diff
@@ -120,8 +120,8 @@ H5_trace(const double *returning, const char *func, const char *type, ...)
     void *            vp                  = NULL;
     FILE *            out                 = H5_debug_g.trace;
     static hbool_t    is_first_invocation = TRUE;
-    H5_timer_t        function_timer      = {{0}, {0}, {0}, FALSE};
-    H5_timevals_t     function_times;
+    H5_timer_t        function_timer;
+    H5_timevals_t     function_times = {0.0, 0.0, 0.0};
```
Comment (bmribler) [inline src/H5trace.c:124]: Hmm, I'm confused.  I thought they need to line up...

File: src/H5trace.c [hunk 5]
```diff
@@ -156,11 +160,10 @@ H5_trace(const double *returning, const char *func, const char *type, ...)
         H5_timer_init(&running_timer);
```
Comment (bmribler) [inline src/H5trace.c:160]: So, in some condition, the timer is initialized again?  Is that ok?

File: src/H5trace.c (hunk 4 above)
Comment (byrnHDF) [reply src/H5trace.c:124]: They would if you swap 123 and 124 because 123 breaks the continuous lineup.

File: src/H5trace.c (hunk 5 above)
Comment (derobins) [reply src/H5trace.c:160]: H5_timer_init() just resets the timer fields.

Comment (jhendersonHDF) [reply src/H5trace.c:160]: This one tripped me up when I first read it. Note that there's a `running_timer` and a `function_timer`

File: src/H5trace.c (hunk 4 above)
Comment (derobins) [reply src/H5trace.c:124]: Also, we let the formatter deal with this.

**Review by derobins — APPROVED** [review]

**Pull request merged**